### PR TITLE
refactor: Deployment script cleanup

### PR DESCRIPTION
**Pull request by pxrl**
- Replace numeric chain IDs with symbolic IDs.
- Define constants for quote timestamp buffer and fill deadline buffer.
- Defer to upstream-defined token addresses.
- General minor refactoring to shrink code.

### CODE REVIEW

**Review by dohaki — APPROVED** [review]
LGTM

File: deployments/deployments.json
Comment (james-a-morris) [inline deployments/deployments.json:1]: Should we have this diff in the PR if the purpose is to swap out constants?

Comment (nicholaspai) [reply deployments/deployments.json:1]: yeah a bit cleaner to remove in a separate PR if you don't mind Paul

**Review by nicholaspai — APPROVED** [review]

File: deployments/deployments.json
Comment (pxrl) [reply deployments/deployments.json:1]: Pulled it out and did some additional cleanup over here: https://github.com/across-protocol/contracts/pull/579

File: deploy/003_deploy_optimism_spokepool.ts [hunk 1]
```diff
@@ -22,13 +23,13 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    "0x4200000000000000000000000000000000000006",
+    WETH[spokeChainId],
     3600,
     21600,
-    L2_ADDRESS_MAP[spokeChainId].l2Usdc,
+    USDC[spokeChainId],
     L2_ADDRESS_MAP[spokeChainId].cctpTokenMessenger,
   ];
-  await deployNewProxy("Optimism_SpokePool", constructorArgs, initArgs, spokeChainId === 10);
+  await deployNewProxy("Optimism_SpokePool", constructorArgs, initArgs, spokeChainId === CHAIN_IDs.OPTIMISM);
```
Comment (pxrl) [inline deploy/003_deploy_optimism_spokepool.ts:32]: Reminder to revert this change because it's no longer desirable to do this after https://github.com/across-protocol/contracts/pull/577

Comment (pxrl) [reply deploy/003_deploy_optimism_spokepool.ts]: Done.

File: deploy/003_deploy_optimism_spokepool.ts [hunk 2]
```diff
@@ -15,17 +15,11 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     hubPool.address,
   ];
 
-  // Construct this spokepool with a:
-  //    * A WETH address of the WETH address
-  //    * A depositQuoteTimeBuffer of 1 hour
-  //    * A fillDeadlineBuffer of 6 hours
-  //    * Native USDC address on L2
-  //    * CCTP token messenger address on L2
```
Comment (pxrl) [inline deploy/003_deploy_optimism_spokepool.ts:23]: I've removed each of these for two reason:
- The input arguments are almost always symbolically named now.
- This comment is liable to get out of sync with the actual contract initializer. It's always better to check against the official source.

**Review by bmzig — APPROVED** [review]

**Review by nicholaspai — APPROVED** [review]
